doc/usecases/map_undo_redo.md

```diff
@@ -27,3 +27,4 @@
 - **Non-functional Constraints:**
   - The app must clearly communicate to the user whether an action can be undone or redone.
   - The app must clearly communicate which undo/redo action was done.
+  - Must also work on concurrent use of the same map (users can undo what others did).
```

Comment:
Is this really wanted? I imagine this could be very confusing.
How do other collaborative tools handle this? E.g. Figma or Google Docs

Reply:
I also think that this could be quite confusing. 

> How do other collaborative tools handle this? E.g. Figma or Google Docs

It tested it in Google Docs with another person.
The undo redo history is saved in memory locally, so closing and reopening the browser prevents any further undo or redo.
Undo and redo with <kbd>ctrl</kbd> + <kbd>z</kbd> only affect the parts that got written by the user herself, but there are also document versions, which get saved periodically.

Reply:
Yes, I agree. Let us do undo/redo user specific. The rest of the text actually already was like this.
We still have the "versioning" use case for the concurrent use, I added the non-functional constraint there.

It would be interesting how others do it but it must some solution that actually has server-side undo.
Etherpad Lite has such a solution, called timeslider: https://yopad.eu/p/sdsdw/timeslider#4
But you are right, also Etherpad Lite's "normal" undo is specific to a user/session (didn't check how it works for an logged-in user)

@Bushuo can you include in your offline feasibility study the local vs. server-sideundo/redo functionality? This is the most urgent use case to know, as this **heavily** influences the database. There are actually 4 possibilities:
1. undo is local only (change browser etc. and undo is lost)
2. undo is both local and server-side (so also undo history would need to sync after offline case)
3. undo is only server-side (during offline no undo is possible, the whole sync when going offline would be one undo step)
4. undo is only server-side and disabled when offline (simplification of 3.)